pilot/pkg/security/model/authentication.go

```diff
@@ -72,6 +73,9 @@ var SDSAdsConfig = &core.ConfigSource{
 	ResourceApiVersion: core.ApiVersion_V3,
 }
 
+// Holds frequently used secret config to avoid excessive memory allocations.
+var sdsSecretConfigCache = lru.New(100)
```

Comment:
I donot think caching a simple struct which can be built directly, like a construction function. Could help much. 
Actually, we have so many such small points that can be optimized, but introducing cache anywhere is not good AFAIK. Most obviously, it will make the project difficult to read and maintain.

Reply:
The point here, it is built over and over again with the same values similar to "default" and "RootCA" for metadata certs  - The same logic that we applied for "default" and "RootCA" (we cache them) applies for metadata certs as well as it is being built for every cluster and every listener.  I tried to make this generic by introducing cache here - but if you have concerns, I can limit it to just metadata certs similar to "default" and "RootCA" - but IMO that will be even more specific.